benchmarks/testminer.py

```diff
@@ -638,10 +638,12 @@ def get_result_data(self, test_job_id):
         return (json_attachments[0][0], base64.b64decode(json_attachments[0][1]))
 
     def get_environment_name(self, metadata):
-        wanted = ('device', 'mode', 'core')
-        data = [ str(metadata[key]) for key in wanted if key in metadata ]
-        if len(data) == len(wanted):
-            return '-'.join(data)
+        wanted = ('device', 'mode', 'core', 'compiler-mode')
```

Comment:
I'm not that pessimistic and settings can be changed per each instance while the core code can't (easily). This code probably won't make any harm to android-qa-reports as the environment will be set to None. But it's nice to have some options rather than not have them :)